File: cs214/labs/07/NameTester.rb
# NameTester.rb tests class Name and its operations
#
# Begun by: Dr. Adams, for CS 214 at Calvin College.
# Completed by: Advait Scaria
# Date: 03/28/2019
####################################################

require 'test/unit/assertions'   # needed for assert
include Test::Unit::Assertions

class Name 
	
	###############################################################
	# initialize() constructs a Name object from three strings. 
	# Receive: first, middle and last, three strings.
	# Return: the Name, its fields set to first, middle and last. 
	###############################################################

	def initialize(first, middle, last)
       @first, @middle, @last = first, middle, last
    end

    ###############################################################
	# getters/accessors for attributes first, middle and last in Name object.
	###############################################################
    attr_reader :first, :middle, :last

	###############################################################
	# fullName() returns a String containing the first, middle, and
	# 	 		 last names in that order separated by spaces. 
	# PRE: Name object must be initialized.
	# Return: a String representation of the Name object. 
	###############################################################

    def fullName()
    	@first + " " + @middle + " " + @last
    end

	###############################################################
	# print() displays a Name object. 
	# PRE: Name object must be initialized.
	# Output: a String representation of the Name object to the screen.
	# Return: a String representation of the Name object. 
	###############################################################

    def print()
    	puts fullName()
    	return fullName()
    end
end


def testName
   name = Name.new("John", "Paul", "Jones")

   assert name.first == "John", "first failed"
   assert name.middle == "Paul", "middle failed"
   assert name.last == "Jones", "last failed"
   assert name.fullName == "John Paul Jones", "fullName failed"
   assert name.print == "John Paul Jones", "print failed"
   
   print "All tests passed!\n"
end

testName

File: cs214/labs/07/NameTester.java
/* NameTester.java is a 'driver' to test class Name.
 * Begun by: Prof. Adams, for CS 214 at Calvin College.
 * Student: Advait Scaria
 * Date: 03/28/2019
 ************************************************************/

public class NameTester  
{
  public static void main(String[] args)
  {
      Name aName = new Name("John", "Paul", "Jones");

	  assert aName.getFirst().equals("John") ;
      assert aName.getMiddle().equals("Paul") ;
      assert aName.getLast().equals("Jones") ;

      System.out.println(aName);
      assert aName.toString().equals("John Paul Jones");

      System.out.println("All tests passed!");
  }
}

class Name
   {
      private String myFirst,
                     myMiddle,
                     myLast;

      /* Name explicit constructor...constructs a Name object from three strings.
       * Receive: first, middle and last, three strings.
       * POST: Makes a Name object and initializes its three variables to the given parameters.
       */
      public Name(String first, String middle, String last)
   		{
      		myFirst = first;
      		myMiddle = middle;
      		myLast = last;
   		}

   	  /* getFirst()
   	   * PRE: Name is initialized with its three name parts.
       * Return: myFirst, the first name of the Name object.
       */
   	  public String getFirst()
   		{
      		return myFirst;
   		}

      /* getMiddle()
   	   * PRE: Name is initialized with its three name parts.
       * Return: myMiddle, the middle name of the Name object.
       */
   	  public String getMiddle()
   		{
      		return myMiddle;
   		}

   	  /* getLast()
   	   * PRE: Name is initialized with its three name parts.
       * Return: myLast, the last name of the Name object.
       */
   	  public String getLast()
   		{
      		return myLast;
   		}

   	  /* toString()
   	   * PRE: Name is initialized with its three name parts.
       * Return: a String representation of the Name object
       *		 that containts all of its three names together
       *		 separated by spaces.
       */
   	  public String toString()
   		{
      		return myFirst + ' ' + myMiddle + ' ' + myLast;
   		}
   };
